include/simdjson/generic/ondemand/object-inl.h

```diff
@@ -39,6 +39,37 @@ simdjson_inline simdjson_result<value> object::operator[](const std::string_view
 simdjson_inline simdjson_result<value> object::operator[](const std::string_view key) && noexcept {
   return std::forward<object>(*this).find_field_unordered(key);
 }
+#ifdef __cpp_concepts
+template <typename... StrT>
+    requires details::all_string_views<StrT...>
+[[nodiscard]] simdjson_inline std::array<simdjson_result<value>, sizeof...(StrT)> object::find_all(StrT... keys) & noexcept {
+  using array_type = std::array<simdjson_result<value>, sizeof...(StrT)>;
+  array_type values;
+  struct appender {
+    simdjson_result<field> value;
+    array_type& values;
+    std::size_t index = 0;
+
+    appender& operator|=(std::string_view const key) noexcept {
+      if (value.key() == key) {
+        values[index] = std::move(value).value();
+      }
+      ++index;
+      return *this;
+    }
+  };
+  for(auto value : *this) {
+    (appender{value, values} |= ... |= keys);
+  }
+  return values;
+}
+
+template <typename... StrT>
+    requires details::all_string_views<StrT...>
+[[nodiscard]] simdjson_inline std::array<simdjson_result<value>, sizeof...(StrT)> object::find_all(StrT... keys) && noexcept {
```

Comment:
Can you tell me more about the [[nodiscard]] usage here. Why is it a problem if the return of this function is not actually used? 

@the-moisrex

Reply:
@FranciscoThiesen, Is there a reasonable place where someone would call `.find_all` and just ignore the result? If that happens, it's pretty much a bug, and `[[nodiscard]]` helps catch those bugs; and if they actually need to do that, it's better if they use `std::ignore = ....find_all(...)` to **state the intent** for everyone.

We could actually add `[[nodiscard]]` to a lot of places in this library, but since it's a C++11 library, we either have to get macro-creative, or just not do it; but there's no reason for a C++20-only part of the library not to have it.

Same goes for `constexpr` btw. We can make a lot of machinery in the library `constexpr`-friendly.